src/ProjectNeon/Assets/Features/Effects/HealFlatForTurnsOnTurnStart.cs

```diff
@@ -0,0 +1,16 @@
+﻿public sealed class HealFlatForTurnsOnTurnStart : Effect
+{
+    private int _turns;
+    private int _amount;
+
+    public HealFlatForTurnsOnTurnStart(int amount, int turns) {
+        _turns = turns;
+        _amount = amount;
+    }
+
+    public void Apply(Member source, Target target)
```

Comment:
For `Regen` itself, the caster who applied the effect is not the one who is doing the healing. The character with Regen heals himself.

Reply:
@SilasReinagel `Heal` effect doesn't use the caster info. The target always heals itself:
```
    {
        target.Members.ForEach(
            member => {
                member.State.GainHp(_amount);
            }
        );
```